src/cogent3/core/new_alignment.py

```diff
@@ -198,10 +196,16 @@ def get_seq_bytes(
 
     def get_seq_view(self, seqid: str) -> SeqDataView:
         seq_len = len(self._data[seqid])
-        return SeqDataView(seq=self, seqid=seqid, seq_len=seq_len)
+        return SeqDataView(seqs=self, seqid=seqid, seq_len=seq_len)
+
+    def subset(self, names: Union[str, typing.Sequence[str]]) -> SeqsData:
+        """Returns a new SeqsData object with only the specified names."""
+        names = [names] if isinstance(names, str) else names
```

Comment:
what happens if `names=...` provided do not match `self.names`?

Reply:
`<dict>.get` should handle if the keys are not present. But if the resulting data is an empty dict then we should probs raise an error